File: Patrones/Comportamiento/Strategy/Java/src/PagoConTarjetaDeCredito.java
public class PagoConTarjetaDeCredito implements EstrategiaDePago {
    private String numeroTarjeta;
    private String titular;

    public PagoConTarjetaDeCredito(String numeroTarjeta, String titular) {
        this.numeroTarjeta = numeroTarjeta;
        this.titular = titular;
    }

    @Override
    public void pagar(double pago) {
        System.out.println("Pagando " + pago + " usando tarjeta de crédito. Detalles: " + 
                            "Número de tarjeta: " + numeroTarjeta + ", Titular: " + titular);
    }
}

File: Patrones/Comportamiento/Strategy/Python/PagoConTarjetaDeCredito.py
from EstrategiaDePago import EstrategiaDePago


# Estrategia de pago con tarjeta de crédito
class PagoConTarjetaDeCredito(EstrategiaDePago):

    def __init__(self, numero_tarjeta, titular):
        self.__numero_tarjeta = numero_tarjeta
        self.__titular = titular

    def pagar(self, monto):
        print(
            f"Pagando {monto} usando tarjeta de crédito. Detalles: "
            f"Número de tarjeta: {self.__numero_tarjeta}, Titular: {self.__titular}"
        )
